sync-core/src/test/java/com/cloudant/sync/util/TestUtils.java

```diff
@@ -153,5 +148,27 @@ public static File loadFixture(String fileName){
        }
    }
 
+   public static SQLDatabase getDatabaseConnectionToExsitingDb(SQLDatabase db){
+       if(Misc.isRunningOnAndroid())
+           return db;
+
+       try {
+           String filePath = (String) db.getClass()
```

Comment:
Actually I don't even understand the purpose of this method, why do we need to open a new connection to an existing database (in test code), and why does it just return the current connection on android?